datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -451,48 +456,98 @@ def _send_telemetry_counter(self, metric_name, val, scraper_config, extra_tags=N
                 # but only the active one is set to 1 (others are set to 0)
                 # example: kube_pod_status_phase in kube-state-metrics
                 if sample[self.SAMPLE_VALUE] != 1:
                     continue
-                label_dict = dict()
-                matching_value = None
-                for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
-                    if label_name == matching_label:
-                        matching_value = label_value
-                    elif label_name in scraper_config['label_joins'][metric.name]['labels_to_get']:
-                        label_dict[label_name] = label_value
-                try:
-                    if scraper_config['_label_mapping'][matching_label].get(matching_value):
-                        scraper_config['_label_mapping'][matching_label][matching_value].update(label_dict)
-                    else:
-                        scraper_config['_label_mapping'][matching_label][matching_value] = label_dict
-                except KeyError:
-                    if matching_value is not None:
-                        scraper_config['_label_mapping'][matching_label] = {matching_value: label_dict}
+
+                if 'label_to_match' in label_joins:
+                    label_dict = dict()
+                    matching_value = None
+                    matching_label = label_joins['label_to_match']
+
+                    for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                        if label_name == matching_label:
+                            matching_value = label_value
+                        elif label_name in labels_to_get or '*' in labels_to_get:
+                            label_dict[label_name] = label_value
+
+                    self._update_label_mapping(matching_label, matching_value, label_dict, scraper_config)
+
+                if 'labels_to_match' in label_joins:
+                    matching_labels = sorted(label_joins['labels_to_match'])
+                    if set(matching_labels).issubset(set(sample[self.SAMPLE_LABELS].keys())):
+                        label_dict = dict()
+                        for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                            if label_name in matching_labels:
+                                continue
+                            if label_name in labels_to_get or '*' in labels_to_get:
+                                label_dict[label_name] = label_value
+
+                        mapping_key = ','.join(matching_labels)
+                        matching_values = [sample[self.SAMPLE_LABELS][l] for l in matching_labels]
+                        mapping_value = ','.join(matching_values)
+
+                        self._update_label_mapping(mapping_key, mapping_value, label_dict, scraper_config)
+
+    def _update_label_mapping(self, key, value, label_dict, scraper_config):
```

Comment:
That can be done in a one liner `scraper_config['_label_mapping'][key].setdefault(value, {}).update(label_dict)`